src/chat.rs

```diff
@@ -1786,10 +1948,13 @@ pub async fn delete_device_expired_messages(context: &Context) -> Result<bool, E
             )
             .await?;
 
-        Ok(rows_modified > 0)
-    } else {
-        Ok(false)
+        updated |= rows_modified > 0;
+    }
+
+    if updated {
+        schedule_autodelete_task(context).await;
```

Comment:
I don't get this.  All this `schedule_autodelete_task()` does is send out a `MsgsChanged` event either some time in the future, or right now when there already is a next message to delete.  But why is this being done via a scheduled task?  Didn't we just delete messages right here in this function?  Why is this not sending out the `MsgsChanged` event right away and why do we need to bother with this scheduling?

Reply:
We have just deleted all the messages that are already expired. There may be some messages that will expire in the future, so we schedule autodelete task to trigger UI reload with MsgsChanged at the time of expiration of the next message.